### DISCUSSION TITLE
How to query Nodes in Flatfile

### DISCUSSION THREAD

**Discussion by D3M05** [osm2pgsql-dev/osm2pgsql#1622] [Q&A] [answered]
tl;dr can you query or read the cache flatnodes file to see if a node id exists.

I'm running into a problem with my updates and seeing large geometries dropping.  I'm already running v1.5.1, so the memory problem **should** not be an issue.  I can see that the ways are all there, and can check the polygon, lines, and points table, but was wondering if there was a way to check whether the nodes exist in cache.  That way I have an idea if my manual update program still has a bug in it, or if this is an issue with something else.  I did try re-importing a broken relation with full debugging, and there are no errors.  In it's process it will clear and copy data correctly to the rels table with stdin data, but then will try to copy to roads table but stdin data is empty.  It should also create a record in polygon table, but doesn't attempt it.

Any help in tracking down the problem is greatly appreciated!

**Comment by joto** [comment, marked as answer]
If you have [Osmium](https://osmcode.org/osmium-tool/) installed, you can use [osmium query-locations-index](https://docs.osmcode.org/osmium/latest/osmium-query-locations-index.html) to look into the flat node file.

**Reply by D3M05** [reply]
Thank you!  No idea why I couldn't find this function in my searches.

It takes about 25min to check all the nodes on a relation like 4039900, but will work well to debug my updater.  Super grateful for the help!
